designs/comprehensions-design.md

```diff
@@ -0,0 +1,105 @@
+# Comprehensions Team Design Doc
+
+## Test cases
+
+```
+a : List = None
```

Comment:
I would recommend _not_ using `List` types here. First of all, `List` isn't enough to know the type of the _contents_. Second, you're dependent on the lists group to run anything at all.

I would recommend writing your tests in terms of `()` comprehensions so that you're in total control of the generator/iterator representation (as you discuss below, by creating new classes).